libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -196,32 +202,37 @@ void AP_BattMonitor_Backend::check_failsafe_types(bool &low_voltage, bool &low_c
 
     // check critical battery levels
     if ((voltage_used > 0) && (_params._critical_voltage > 0) && (voltage_used < _params._critical_voltage)) {
-        critical_voltage = true;
+        fs.critical_voltage = true;
     } else {
-        critical_voltage = false;
+        fs.critical_voltage = false;
     }
 
     // check capacity failsafe if current monitoring is enabled
     if (has_current() && (_params._critical_capacity > 0) &&
         ((_params._pack_capacity - _state.consumed_mah) < _params._critical_capacity)) {
-        critical_capacity = true;
+        fs.critical_capacity = true;
     } else {
-        critical_capacity = false;
+        fs.critical_capacity = false;
     }
 
     if ((voltage_used > 0) && (_params._low_voltage > 0) && (voltage_used < _params._low_voltage)) {
-        low_voltage = true;
+        fs.low_voltage = true;
     } else {
-        low_voltage = false;
+        fs.low_voltage = false;
     }
 
     // check capacity if current monitoring is enabled
     if (has_current() && (_params._low_capacity > 0) &&
         ((_params._pack_capacity - _state.consumed_mah) < _params._low_capacity)) {
-        low_capacity = true;
+        fs.low_capacity = true;
     } else {
-        low_capacity = false;
+        fs.low_capacity = false;
     }
+    
+    fs.below_temperature = has_temperature() && !is_equal(_params._minimum_temperature.get(), DEFAULT_FAILSAFE_TEMPERATURE) &&
```

Comment:
```suggestion
    fs.below_temperature = **has_temperature()** && !is_equal(_params._minimum_temperature.get(), DEFAULT_FAILSAFE_TEMPERATURE) &&
```
For `has_temperature()`. What if we have dropped reading of temperature due to a timeout or failure to read?  We will not set an `Unhealthly` state. Because we don't know that pack does `return_temperature` normally?